art/poison_detection/spectral_signature_defense.py

```diff
@@ -0,0 +1,200 @@
+# MIT License
+#
+# Copyright (C) Two Six Labs, LLC 2020
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements methods performing backdoor poisoning detection based on spectral signatures.
+
+| Paper link: https://papers.nips.cc/paper/8024-spectral-signatures-in-backdoor-attacks.pdf
+
+| Please keep in mind the limitations of defenses. For more information on the limitations of this
+    specific defense, see https://arxiv.org/abs/1905.13409 .
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import numpy as np
+
+from art.poison_detection.ground_truth_evaluator import GroundTruthEvaluator
+from art.poison_detection.poison_filtering_defence import PoisonFilteringDefence
+
+
+class SpectralSignatureDefense(PoisonFilteringDefence):
+    """
+    Method from Tran et al., 2018 performing poisoning detection based on Spectral Signatures
+    """
+
+    defence_params = PoisonFilteringDefence.defence_params + [
+        "classifier",
+        "x_train",
+        "y_train",
+        "batch_size",
+        "eps_multiplier",
+        "ub_pct_poison",
+        "nb_classes",
+    ]
+
+    def __init__(self, classifier, x_train, y_train, **kwargs):
+        """
+        Create an :class:`.ActivationDefence` object with the provided classifier.
```

Comment:
Is ActivationDefence still correct?
```suggestion
        Create an :class:`.SpectralSignatureDefense` object with the provided classifier.
```